File: Echowaves/AddChildWaveViewController.m
//
//  AddChildWaveViewController.m
//  Echowaves
//
//  Created by Dmitry on 4/2/14.
//  Copyright (c) 2014 Echowaves. All rights reserved.
//

#import "AddChildWaveViewController.h"

@interface AddChildWaveViewController ()

@end

@implementation AddChildWaveViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    [self createChildWave].layer.cornerRadius = 4.0f;
    [self createChildWave].layer.borderWidth = 1.0f;
    [self createChildWave].layer.borderColor = UIColorFromRGB(0xFFA500).CGColor;
    
    [self childWaveName].layer.cornerRadius = 4.0f;
    [self childWaveName].layer.borderWidth = 1.0f;
    [self childWaveName].layer.borderColor = UIColorFromRGB(0xFFA500).CGColor;
}

- (void) viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    [self.childWaveName becomeFirstResponder];
    self.childWaveName.text = [NSString stringWithFormat:@"%@.", APP_DELEGATE.currentWaveName];
}

- (IBAction)createChildWave:(id)sender {
    [EWWave createChildWaveWithName:[self childWaveName].text
                            success:^(NSString *waveName) {
                                [self.navigationController popViewControllerAnimated:YES];
                            }
                            failure:^(NSString *errorMessage) {
                                [EWWave showErrorAlertWithMessage:errorMessage
                                                  FromSender:nil];
                                [self.navigationController popViewControllerAnimated:YES];

                            }];
}

@end

File: Echowaves/AddChildWaveViewController.swift
//
//  AddChildWaveViewController.swift
//  Echowaves
//
//  Created by D on 12/2/14.
//  Copyright (c) 2014 Echowaves. All rights reserved.
//

import Foundation

class AddChildWaveViewController:UIViewController {
    
    @IBOutlet var childWaveName:UITextField!
    @IBOutlet var createChildWave:UIButton!
    
    @IBAction func createChildWave(sender: AnyObject?) {
        
                EWWave.createChildWave(
                    self.childWaveName!.text,
                    success: { (newWaveName) -> Void in
                        if let navController = self.navigationController {
                            navController.popViewControllerAnimated(true)
                        }
                    },
                    failure:{ (errorMessage) -> Void in
                        EWWave.showErrorAlertWithMessage(errorMessage, fromSender: self)
                        self.navigationController?.popViewControllerAnimated(true)
        })
        
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.createChildWave.layer.cornerRadius = 4.0
        self.createChildWave.layer.borderWidth = 1.0
        self.createChildWave.layer.borderColor = UIColor(rgb:0xFFA500).CGColor
        
        self.childWaveName.layer.cornerRadius = 4.0
        self.childWaveName.layer.borderWidth = 1.0
        self.childWaveName.layer.borderColor = UIColor(rgb:0xFFA500).CGColor
    }
    
    override func viewDidAppear(animated:Bool) {
        super.viewDidAppear(animated)
        self.childWaveName.becomeFirstResponder()
        self.childWaveName.text = "\(APP_DELEGATE.currentWaveName)."
    }
    
}
